public/API/API_GetAchievementOfTheWeek.php

```diff
@@ -82,26 +103,49 @@
     'Title' => $achievementData['ConsoleName'] ?? null,
 ];
 
-$forumTopic = [
-    'ID' => $staticData['Event_AOTW_ForumID'] ?? null,
-];
+if ($achievementOfTheWeek) {
+    $unlocks = collect();
 
-$parentGame = getParentGameFromGameTitle($game['Title'], $achievementData['ConsoleID']);
-$unlocks = getAchievementUnlocksData((int) $achievementID, null, $numWinners, $numWinnersHardcore, $numPossibleWinners, $parentGame?->id, 0, 500);
+    $playerAchievements = $achievementOfTheWeek->achievement->playerAchievements()
+        ->with('user')
+        ->orderBy(DB::raw('IFNULL(unlocked_hardcore_at, unlocked_at)'))
+        ->get();
```

Comment:
If `$achievementOfTheWeek` is achievement ID 9, I get an out of memory error when running this query:

```php
$achievement = Achievement::find(9);

$playerAchievements = $achievement
  ->playerAchievements()
  ->with("user")
  ->orderBy(DB::raw("IFNULL(unlocked_hardcore_at, unlocked_at)"))
  ->get();
```

This is pulling too much data. I think we may need to implement pagination at the endpoint level via `o` (offset) and `c` (count) params.

Reply:
Sorry. I actually think it may be my interpretation of this code that's incorrect.

I think I'm getting confused by the `achievement` and `sourceAchievement` relations. In this case, `sourceAchievement` would be ID 9. `achievement` is the literal event achievement (visible on the game page for the event).

Please disregard this feedback assuming I made a mistake here.

Reply:
It might be worth putting an upper bound on that anyway. The old code had a limit of 500, though the imported data for 2024 shows the most unlocked AotW has 292 entries.